### PR TITLE
CAKE-5358 | Render Affiliate Unit Markup

### PR DESCRIPTION
**Pull request by cschucode**
Using mustache templating engine to create affiliate unit markup.

### CODE REVIEW

File: extensions/wikia/AffiliateService/js/ext.AffiliateService.js [hunk 1]
```diff
@@ -114,13 +116,16 @@ require([
...
+					// add unit data to be inserted into template
+					AffiliateService.addMarker(units[0]);
 				} else {
 					console.log('No units available');
 				}
 			});
 		},
 
-		addMarker: function () {
+		addMarker: function (unit) {
```
Comment (mklucsarits) [inline extensions/wikia/AffiliateService/js/ext.AffiliateService.js]: I think we should rename this function now, I believe it was named literally for the marker Josh was inserting.

File: extensions/wikia/AffiliateService/js/templates.mustache.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+define( 'ext.wikia.AffiliateService.templates', [], function() { 
+    'use strict'; 
+    
+    return {
+        "AffiliateService_unit" : '<div style="display: flex; background: red; width: 100%; height: 100px"><img style="height: 100%;" src="{{image}}"><p>{{heading}}</p><button>{{buttonText}}</button><img src="{{logo}}" /></div>',
```
Comment (mklucsarits) [inline extensions/wikia/AffiliateService/js/templates.mustache.js:5]: Can we use an external file for the HTML and external CSS?

Comment (vforge) [reply extensions/wikia/AffiliateService/js/templates.mustache.js:5]: We can and we should sue external CSS - as for external markdown file I couldn't find any existing examples of that.

Comment (cschucode) [reply extensions/wikia/AffiliateService/js/templates.mustache.js:5]: Using an external `.scss` file in CAKE-5359

File: extensions/wikia/AffiliateService/js/ext.AffiliateService.js (hunk 1 above)
Comment (cschucode) [reply extensions/wikia/AffiliateService/js/ext.AffiliateService.js]: Renamed this to `renderUnitMarkup`.

File: extensions/wikia/AffiliateService/js/units.js [hunk 3]
```diff
@@ -1,22 +1,27 @@
...
 		{
-			category: 'disney2',
-			campaign: 'disneyplus2',
+			category: 'disneyplus',
+			campaign: 'mandalorian',
 			name: 'test unit2',
-			country: ['US', 'CA', 'NL', 'AU', 'NZ'],
-			header: 'Header 2',
-			subheader: 'This is subheader 2',
+			country: ['US'],
```
Comment (vforge) [inline extensions/wikia/AffiliateService/js/units.js]: Please bring back the proper GEO for both units.

**Review by vforge — APPROVED** [review]

File: extensions/wikia/AffiliateService/js/units.js (hunk 3 above)
Comment (cschucode) [reply extensions/wikia/AffiliateService/js/units.js]: Gotcha!

**Pull request merged**